apptest/model.go

```diff
@@ -42,8 +43,23 @@ type PrometheusWriteQuerier interface {
 
 // QueryOpts contains various params used for querying or ingesting data
 type QueryOpts struct {
-	Tenant  string
-	Timeout string
+	Tenant       string
+	Timeout      string
+	ModifyValues func(url.Values)
```

Comment:
I don't think that it's good API. It makes no sense to call `PrometheusAPIV1Query` without `query` param as it's required for it and it will only confuse users. 

 Meanwhile,  start, step, end is optional and can be moved into the opts with explicit names.

Reply:
> I don't think that it's good API. It makes no sense to call `PrometheusAPIV1Query` without `query` param as it's required for it and it will only confuse users.
> 
> Meanwhile, start, step, end is optional and can be moved into the opts with explicit names.

I guess this makes sense. One thing to point out is that the method implementations will then need to:

1. explicitly pick the params that they accept. For example APIV1Query will need to explicitly use opts.Time param while QueryRange will need to explicitly use opts.Start and opts.End param.
2. If the opts param(s) are not set (i.e. they have the default value, which is empty string in most cases), the method should not pass that param to the API. Although this might not be needed if the API can interpret the  empty string as no value.